app/models/report.rb

```diff
@@ -39,4 +39,49 @@ def statuses
   def media_attachments
     MediaAttachment.where(status_id: status_ids)
   end
+
+  def assign_to_self!(current_account)
+    update!(assigned_account_id: current_account.id)
+  end
+
+  def unassign!
+    update!(assigned_account_id: nil)
+  end
+
+  def resolve!(acting_account)
+    update!(action_taken: true, action_taken_by_account_id: acting_account.id)
+  end
+
+  def unresolve!
+    update!(action_taken: false, action_taken_by_account_id: nil)
+  end
+
+  def unresolved?
+    !action_taken?
+  end
+
+  def history
+    report_log = Admin::ActionLog.where(
+      target_type: 'Report',
+      target_id: id,
+      created_at: created_at..updated_at
+    ).unscope(:order)
+
+    target_account_log = Admin::ActionLog.where(
+      target_type: 'Account',
+      target_id: target_account_id,
+      created_at: created_at..updated_at
+    ).unscope(:order)
+
+    statuses_log = Admin::ActionLog.where(
+      target_type: 'Status',
+      target_id: status_ids,
+      created_at: created_at..updated_at
+    ).unscope(:order)
+
+    query = "((#{report_log.to_sql}) UNION ALL (#{target_account_log.to_sql}) UNION ALL (#{statuses_log.to_sql})) as admin_action_logs"
+    sql = Admin::ActionLog.connection.unprepared_statement(query)
```

Comment:
It seems that this method takes a block argument, i.e. `unprepared_statement { query }`, but I am not totally sure why unprepared_statement is needed in here?

Reply:
We may want to just use https://github.com/brianhempel/active_record_union to do this the clean way